testing/trino-faulttolerant-tests/src/test/java/io/trino/faulttolerant/TestOverridePartitionCountRecursively.java

```diff
@@ -0,0 +1,307 @@
+                        " ON n1.nationkey = n2.nationkey + 1",
+                ImmutableMap.<Integer, FragmentPartitioningInfo>builder()
+                        .put(0, new FragmentPartitioningInfo(COORDINATOR_DISTRIBUTION, Optional.empty(), SINGLE_DISTRIBUTION, Optional.empty()))
+                        .put(1, new FragmentPartitioningInfo(SCALED_WRITER_ROUND_ROBIN_DISTRIBUTION, Optional.empty(), SINGLE_DISTRIBUTION, Optional.empty()))
+                        .put(2, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.empty(), SCALED_WRITER_ROUND_ROBIN_DISTRIBUTION, Optional.empty()))
+                        .put(3, new FragmentPartitioningInfo(SOURCE_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .put(4, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .put(5, new FragmentPartitioningInfo(SOURCE_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .put(6, new FragmentPartitioningInfo(SOURCE_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .buildOrThrow(),
+                ImmutableMap.<Integer, FragmentPartitioningInfo>builder()
+                        .put(0, new FragmentPartitioningInfo(COORDINATOR_DISTRIBUTION, Optional.empty(), SINGLE_DISTRIBUTION, Optional.empty()))
+                        .put(1, new FragmentPartitioningInfo(SCALED_WRITER_ROUND_ROBIN_DISTRIBUTION, Optional.empty(), SINGLE_DISTRIBUTION, Optional.empty()))
+                        .put(2, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.of(40), SCALED_WRITER_ROUND_ROBIN_DISTRIBUTION, Optional.empty()))
+                        .put(3, new FragmentPartitioningInfo(SOURCE_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .put(4, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.of(40), FIXED_HASH_DISTRIBUTION, Optional.of(40)))
+                        .put(5, new FragmentPartitioningInfo(SOURCE_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .put(6, new FragmentPartitioningInfo(SOURCE_DISTRIBUTION, Optional.empty(), FIXED_HASH_DISTRIBUTION, Optional.empty()))
+                        .put(7, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.of(5), FIXED_HASH_DISTRIBUTION, Optional.of(40)))
+                        .put(8, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.of(5), FIXED_HASH_DISTRIBUTION, Optional.of(40)))
+                        .put(9, new FragmentPartitioningInfo(FIXED_HASH_DISTRIBUTION, Optional.of(5), FIXED_HASH_DISTRIBUTION, Optional.of(40)))
+                        .buildOrThrow(),
+                ImmutableSet.of(3, 5, 6));
+    }
+
+    @Test
+    public void testSkipBroadcastSubtree()
```

Comment:
nit: is this test specifically about testing of skipping broadcast subtrees, or it generally tests rewriter mechanism. Looks like the latter. Change the name?

Reply:
It's actually both. For `testCreateTableAs` it's the same, it also generally tests the rewriter mechanism.